### ISSUE TITLE
Style/CombinableLoops false trigger

### ISSUE THREAD

**Issue by bagedevimo** [rubocop/rubocop#11474]
When two loops are looping over the same enumerable but forming operations on differing objects inside the loop, CombinableLoops fires suggested they be combined which would result in different program behaviour.

--------

## Expected behavior

Rubocop should not flag the loops as combinable

## Actual behavior

Rubocop is flagging the two loops as cominable, even though combing them would result in a differing program behaviour.

## Steps to reproduce the problem

```
$ mkdir example && cd example
$ bundle init
$ bundle add rubocop
$ cat >> rubocop.rb

# frozen_string_literal: true

numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

array_a = []
array_b = []

numbers.shuffle.take(5).each do |n|
  array_a << n
end

numbers.shuffle.take(5).each do |n|
  array_b << n
end


$ bundle exec rubocop ./example.rb
Inspecting 1 file
C

Offenses:

example.rb:12:1: C: Style/CombinableLoops: Combine this loop with the previous loop.
numbers.shuffle.take(5).each do |n| ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

## RuboCop version
```
 bundle exec rubocop -V
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.2) [x86_64-darwin21]
```

**Comment by vlad-pisanov** [comment]
This cop is marked as Unsafe for this reason. Auto-correct at your own risk 😉
